### PR TITLE
neovim: update to 0.9.0

### PR DESCRIPTION
**Pull request by ibhagwan**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

**Comment by ibhagwan** [comment]
> I'd either pull the tree-sitter update into this PR, or mark this PR as draft until the tree-sitter PR is merged.

https://github.com/void-linux/void-packages/commit/346883f9202e374a4d1a4f99c849dd21e569b28b

**Comment by mhmdanas** [comment]
It's okay to update multiple packages in one PR, but each update should go in a separate commit, not in one commit like you've done here.

**Comment by ibhagwan** [comment]
@mhmdanas, done, separated commits and closed #43325.

File: srcpkgs/tree-sitter/template [hunk 1]
```diff
@@ -9,12 +9,12 @@ maintainer="Érico Nogueira <ericonr@disroot.org>"
...
 homepage="https://tree-sitter.github.io"
 distfiles="https://github.com/tree-sitter/tree-sitter/archive/v${version}.tar.gz"
-checksum=b355e968ec2d0241bbd96748e00a9038f83968f85d822ecb9940cbe4c42e182e
+checksum=6181ede0b7470bfca37e293e7d5dc1d16469b9485d13f13a605baec4a8b1f791
 make_check=no # tests require generating fixtures from remote repositories
 
 post_patch() {
 	# fixes an indexmap error when cross compiling
-	cargo update --package autocfg:1.0.1 --precise 1.1.0
+	cargo update --package autocfg:1.1.0 --precise 1.1.0
```
Comment (classabbyamp) [inline srcpkgs/tree-sitter/template:17]: updating from 1.1.0 to 1.1.0?

Comment (icp1994) [reply srcpkgs/tree-sitter/template:17]: none of that is needed anymore, see https://github.com/void-linux/void-packages/pull/43292

Comment (ibhagwan) [reply srcpkgs/tree-sitter/template:17]: ty @icp1994, should I remove the tree-sitter commit from this PR altogether?

**Comment by ibhagwan** [comment]
Removed the tree-sitter update since #43292.

**Comment by paper42** [comment]
don't remove it from this PR, include that commit and link the PR

**Comment by ibhagwan** [comment]
> don't remove it from this PR, include that commit and link the PR

Done, removed the `post patch` step as suggestion by @icp1994.

**Comment by icp1994** [comment]
reading some upstream discussion, you need to add `-DCI_BUILD=OFF` in `configure_args`

**Comment by ibhagwan** [comment]
> reading some upstream discussion, you need to add `-DCI_BUILD=OFF` in `configure_args`

To neovim or tree-sitter? Which line would that be?

**Comment by ibhagwan** [comment]
> reading some upstream discussion, you need to add `-DCI_BUILD=OFF` in `configure_args`

Done, rebuilt both tree-sitter and neovim, seems to work fine.

Don't see any difference in `:version` or binary size btw.

**Comment by icp1994** [comment]
Needs another line in pre_configure for cross-compiling
```
pre_configure() {
	vsed -i runtime/CMakeLists.txt \
		-e "s|\".*/bin/nvim|\${CMAKE_CROSSCOMPILING_EMULATOR} &|g"
	vsed -i src/nvim/po/CMakeLists.txt \
		-e "s|\$<TARGET_FILE:nvim|\${CMAKE_CROSSCOMPILING_EMULATOR} &|g"
}
```

**Comment by ibhagwan** [comment]
Ty @icp1994, all build checks seems to pass now.

https://github.com/void-linux/void-packages/pull/43326/commits/dd905045e6ca4481135137ec44a9cd262c025fca

**Comment by mhmdanas** [comment]
Tested this locally, don't see any issues.

**Pull request merged**